packages/rollup-core/src/app/data/consumers/canonical-chain-batch-submitter.ts

```diff
@@ -91,13 +91,18 @@ export class CanonicalChainBatchSubmitter extends ScheduledTask {
     try {
       const txsCalldata: string[] = this.getTransactionBatchCalldata(l2Batch)
 
+      // TODO: update this to work with geth-persisted timestamp/block number that updates based on L1 actions
       const timestamp = l2Batch.transactions[0].timestamp
+      const blocknumber =
+        (await this.canonicalTransactionChain.provider.getBlockNumber()) - 10 // broken for any prod setting but works for now
+
       log.debug(
         `Submitting tx batch ${l2Batch.batchNumber} with ${l2Batch.transactions.length} transactions to canonical chain. Timestamp: ${timestamp}`
       )
       const txRes: TransactionResponse = await this.canonicalTransactionChain.appendSequencerBatch(
         txsCalldata,
-        timestamp
+        timestamp,
+        blocknumber
```

Comment:
Will there need to be a corresponding change in `log-handlers.ts` for this or any other changes in this PR or does this not change the data structure at all?

Reply:
Hmmm, I did leave the handlers alone.  Mainly this was because the handlers are already ignoring the l2 timestamp, so I figured we had a separate ticket for it.  Talking with @karlfloersch about it right now, will make a ticket to do this if it's not already represented.